### PR TITLE
javascript: Improve gzip/deflate artifacts from ReleaseFiles

### LINKED ISSUE #4566: Support upload of GZIPped artifacts through the API
Hi, would it be possible to upload GZIPped artifacts through the sentry API when the upload is done with the Header `Content-Encoding: gzip`?

Basically if that header is present during upload Sentry would unzip the received artifact after the upload is complete (ex: a sourcemap) and keep the same name.

### PR DESCRIPTION
**Pull request by mattrobenolt**
* Don't gzip files before caching. This is unnecessary because python-memcached will gzip for us. By doing this ourselves, we're actually adding extra work by double gzipping.
* Handle artifacts uploaded as gzip or deflate.

Fixes GH-4566

/cc @benvinegar

### CODE REVIEW

File: tests/sentry/lang/javascript/test_processor.py [hunk 1]
```diff
@@ -72,6 +73,126 @@ def test_unicode(self):
...
+            organization_id=project.organization_id,
+            version='abc',
+        )
+        release.add_project(project)
+
+        file = File.objects.create(
+            name='file.min.js',
+            type='release.file',
+            headers={
+                'Content-Type': 'application/json; charset=utf-8',
```
Comment (mitsuhiko) [inline tests/sentry/lang/javascript/test_processor.py]: going to be a party pooper but 'application/json' does not accept a charset :P

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

**Review by mitsuhiko — APPROVED** [review]

**Comment by mitsuhiko** [comment]
The comment should not require a change to this PR. But we should not do this mistake.

**Comment by benvinegar** [comment]
‼️

**Comment by mattrobenolt** [comment]
Had to revert because things slowed down pretty drastically and not clear what the cause is without some investigation.

**Comment by STRML** [comment]
Is this line in the 8.12 changelog inaccurate, then?

> Allow gziped/deflated JavaScript artifacts to be uploaded through the API.

**Comment by mattrobenolt** [comment]
Oh. Wups. Yes, this is inaccurate. The code landed, then had to be reverted. I'll fix this up. 

--
Matt Robenolt
@mattrobenolt




On Thu, Jan 26, 2017 at 5:36 AM -0800, "Samuel Reed" <notifications@github.com> wrote:












Is this line in the 8.12 changelog inaccurate, then?



Allow gziped/deflated JavaScript artifacts to be uploaded through the API.

**Comment by lele85** [comment]
Any news on that?

**Comment by deepflame** [comment]
@mattrobenolt is this issue fixed and deployed?

**Comment by corydolphin** [comment]
Any update? This would be a big improvement for my team, time wise. We have ~30mb of sourcemaps, which is quite slow to upload.

**Comment by pierreant-p** [comment]
Same here :) (100+ mo of sourcemaps :) )

**Pull request merged**